src/commands/mod.rs

```diff
@@ -726,6 +729,76 @@ struct RestoreArgs {
     revision: Option<RevisionArg>,
 }
 
+/// The Strategy `run` uses to deal with any failure in it's subprocesses.
+// TODO: Move to run.rs
+#[derive(Debug, Default, Clone, clap::ValueEnum)]
+enum ErrorStrategy {
+    /// Stop at any revision if the subprocess fails.
+    /// Cancels any scheduled work but lets already running processes finish
+    /// their work.
+    #[default]
+    Stop,
+    /// Continue if an error occurs, similar to Make's --keep-going.
+    Continue,
+    /// Signal a fatal failure and stop immediately any processes. Also stops
+    /// any running jobs.
+    Fatal,
+}
+
+/// Run a command across a set of revisions.
+///
+///
+/// All recorded state will be persisted in the `.jj` directory, so occasionally
+/// a `jj run --clean` is needed to cleanup disk space.
+///
+/// # Example
+///
+/// # Run pre-commit on your local work
+/// $ jj run 'pre-commit.py .github/pre-commit.yaml' -r (main..@) -j 4
+///
+/// This allows pre-commit integration and other funny stuff.
+#[derive(clap::Args, Clone, Debug)]
+#[command(verbatim_doc_comment)]
+struct RunArgs {
+    /// The command to run across all selected revisions.
+    #[arg(long, short, alias = "x")]
+    command: String,
+    /// The revisions to change.
+    #[arg(long, short, default_value = "@")]
+    revisions: Vec<RevisionArg>,
+    /// How many processes should run in parallel, uses by default all cores.
+    #[arg(long, short)]
+    jobs: usize,
+    /// Setup the required environment and instead of applying running script,
+    /// display all actions which would be done.
+    #[arg(long, short)]
+    dry_run: bool,
+    /// Ignore changes across multiple run invocations.
+    #[arg(long)]
+    read_only: bool,
+    /// Show the diff of an affected revision.
+    #[arg(long)]
+    show: Option<RevisionArg>,
+    /// After running the command rebase the revisions parent onto the new
+    /// commit. This is the default strategy.
+    #[arg(long)]
+    rebase: bool,
+    /// After running the command, re-adjust the parent commit to the new
+    /// revision.
+    #[arg(long, conflicts_with = "rebase")]
+    reparent: bool,
+    /// Clean up all hidden state for run.
+    #[arg(long)]
+    clean: bool,
+    /// Run the command for all revisions, even if a single one fails.
+    /// Implies error-strategy=continue
+    #[arg(short, long)]
+    keep_going: bool,
+    /// The strategy `jj run` should use, if a failure occurs.
+    #[arg(value_enum, default_value = "ErrorStrategy::Stop")]
+    error_strategy: ErrorStrategy,
```

Comment:
I'd rather leave this (and `ErrorStrategy` above) out until we beed them.

Reply:
Agreed, it would be easier to review each option as its functionality is added.

Reply:
Ok, I'll add them with each upcoming feature.